velocidi-sdk/src/main/kotlin/com/velocidi/Util.kt

```diff
@@ -81,4 +83,28 @@ internal object Util {
         parameters.forEach { (k, v) -> builder.appendQueryParameter(k, v) }
         return builder.build()
     }
+
+    fun JSONObject.toQueryParams(): Map<String, String> {
+        fun toQueryParamsAux(elem: Any?, qs: MutableMap<String, String>, path: String) {
+            when (elem) {
+                is JSONObject ->
+                    for (key in elem.keys()) {
+                        val k = if (path.isEmpty()) key else "[$key]"
+                        toQueryParamsAux(elem[key], qs, path + k)
+                    }
+                is JSONArray ->
+                    for (idx in 0..elem.length()) {
+                        val k = if (path.isEmpty()) idx else "[$idx]"
+                        toQueryParamsAux(elem.opt(idx), qs, path + k)
+                    }
+                is Boolean, is Int, is Byte, is Char, is String, is Double, is Float, is Long, is Short ->
```

Comment:
Is this `when` block exhaustive? If not, can't we just have a `else` or default case here?

Reply:
It is not. Unmatched types, like `Null` will not be serialized. In any case, I added an "else case" in 76c5c12 to make this behavior more explicit

Reply:
What do you think of something like this:

```kotlin
            when (elem) {
                JSONObject.NULL, null ->
                    {}
                is JSONObject ->
                    for (key in elem.keys()) {
                        val pathKey = if (path.isEmpty()) key else "[$key]"
                        toQueryParamsAux(elem[key], qs, path + pathKey)
                    }
                is JSONArray ->
                    for (idx in 0..elem.length()) {
                        toQueryParamsAux(elem.opt(idx), qs, "$path[$idx]")
                    }
                else ->
                    qs[path] = elem.toString()
            }
        }
```

Reply:
That change will make us serialize non-primitive types and we might get some weird results. I added an example in 15d19b8 to make this clearer.  Using the implementation above `Foo` would be serialized into `foo=Foo(bar=bar)`, which is not a valid structure for the CDP.